Save power by reducing CPU clock speed of ESP32-C3

ypid opened the pull request:
Tested on:

* 2 times I-9PSL-DE: AirGradient ONE 9th Generation. Total test time: 4 months. No issues.
* 1 time O-1PST: AirGradient Open Air 1st Generation. Total test time: 5 days. No issues.

MallocArray commented:
I think I would likely prefer this be a separate package that users could include if they want to go with the power saving route, but not necessarily change this for all users.

Thoughts?

ypid commented:
I am running 4 ESPHome devices for many months already at 80 MHz without issues. I have observed the log of all of those different device types and might also have checked loop time over a few days. I had no issue yet. One of those 4 devices is a AirGradient One. I have no long term experience running this on the Open Air yet because I just recently got my first. But I don’t expect issues. I feel confident to introduce this as opt-out as it saves power with no downsides. But I am fine with making it opt-in to allow more testers to try it before pushing this to everybody. I have updated this pull request accordingly.

I have updated my initial post to reflect how long I have been testing with 80 MHz already on AirGradient devices.

ypid commented:
> saves power with no downsides

I expected the loop time to increase when uploading results to AirGradient because of TLS so I did a test:

![Screenshot from 2025-04-21 14-45-54](https://github.com/user-attachments/assets/99f07556-2d5e-497a-a86b-5ba1062befb0)

Git Ref "a38d289" was build with 80 MHz. The others with 160 MHz. I have set the upload interval to 10 seconds and only shortly enabled it.

I have also tried to measure power draw.

According to https://www.youtube.com/watch?v=JFDiqPHw3Vc&t=9m43s the ESP32 (not C3) alone with Wifi:

* 160 MHz consumes 130 mA.
* 80 MHz consumes 120 mA.

The most accurate power meter I have on hand is the Riden TC66C. I get I-9PSL-DE, AirGradient ONE 9th Generation, sensors, LED combo and display on:

* 160 MHz: Between 180 to 320 mA
* 80 MHz: Between 175 to 300 mA

I used the Android app to read those numbers from the graph. Pretty amateur like. I guess when I spend this much time on it I should just buy a Power Profiler Kit II (PPK2).

Of course this lower consumption might mess with the temperature calibration of the Open Air. Lets discuss this here: https://github.com/airgradienthq/arduino/issues/305
